proving/proving.go

```diff
@@ -271,8 +275,18 @@ func (p *Prover) tryNonces(numLabels uint64, challenge Challenge, startNonce, en
 		i := i
 		go func() {
 			nonce := startNonce + i
-			indices, err = p.tryNonce(ctx, numLabels, challenge, nonce, workersChans[i], difficulty)
-			resultsChan <- nonceResult{nonce, indices, err}
+
```

Comment:
I'm not sure I understand why these new steps are needed.

Reply:
Accessing the `indices` variable outside the goroutine requires a mutex for both reading a writing. Otherwise. there's a data race. However, it may be moved inside the goroutine, so there's no need for mutexes.

Reply:
Actually, I see that it was already removed by Dmitry in his PR https://github.com/spacemeshos/post/pull/52